src/apps/lwaftr/ipv6_apps.lua

```diff
@@ -147,3 +150,49 @@ function Fragmenter:push ()
       end
    end
 end
+
+-- TODO: handle any NS retry policy code here
+function NDP_eth_next_hop:new(conf)
+   local o = setmetatable({}, {__index=NDP_eth_next_hop})
+   o.conf = conf
+   -- TODO: verify that the src and dst ipv6 addresses and src mac address
+   -- have been provided, in pton format.
+   o.ns_pkt = ndp.form_ns(o.aftr_mac_b4_side, o.aftr_ipv6_ip, o.next_hop_ipv6_addr)
+   o.first_run = true
+   o.resolved = false
+   o.dst_eth = nil
+   return o
+end
+
+function NDP_eth_next_hop:push()
+   local isouth, osouth = self.input.south, self.output.south
+   local inorth, onorth = self.input.north, self.output.north
+   if o.first_run then
+      o.first_run = false
+      transmit(onorth, o.ns_pkt)
+   end
+
+   for _=1,math.min(link.nreadable(isouth), link.writeable(onorth)) do
```

Comment:
Following https://github.com/SnabbCo/snabbswitch/issues/656 we should process all readable packets.